### ISSUE TITLE
Couchbaselite.framework has transitive dependencies

### ISSUE THREAD

**Issue by CharlesGrimont** [couchbase/couchbase-lite-ios#1811]
I'am actually trying to add pod as a dependency of my podspec and I'am not able to push my pod cause of 
transitive dependencies linked to couchbaselite ....

as shown in the error message below

The 'Pods-App' target has transitive dependencies that include static binaries: (/private/var/folders/tr/836cbwq93j1bvy_cg_6t1x780000gn/T/CocoaPods/Lint/Pods/couchbase-lite-ios/Extras/libCBLJSViewCompiler.a, /private/var/folders/tr/836cbwq93j1bvy_cg_6t1x780000gn/T/CocoaPods/Lint/Pods/couchbase-lite-ios/CouchbaseLite.framework, and /private/var/folders/tr/836cbwq93j1bvy_cg_6t1x780000gn/T/CocoaPods/Lint/Pods/couchbase-lite-ios/CouchbaseLiteListener.framework) 

Is there any way to provide a package of couchbaselite-ios that is not including transitive dependencies ? 

Thanks in advance....

--------
* Version:   1.4.0
* Client OS: ios swift

**Comment by pasin** [comment]
For backward compatibility support with the older version of the iOS, we still need to have CouchbaseLite.framework as a static binary. Once we drop the support, we could move to the dynamic framework.

CC: @djpongh

**Comment by CharlesGrimont** [comment]
The thing that i cant understand is that I'm abble to fetch Dynamic frameworks and couchbaselite in a podfile, however i'm not abble to throw a podspec containing both.
I must open an issue with cocoapod.

Have you any Idea  of when will you drop this support ? 
Or create another dynamic branch ? 

I've already Seen couchbaselite-ios2 written un swift that will in the future solve my issue ....

Is there any possibility that i wrap myself couchbaselite into à Dynamic framework and republish that in another pod repo ?

Thanks a lot for your reactivity !

**Comment by pasin** [comment]
You can use Carthage to build a dynamic binary for iOS. From there, you can use the dynamic binary  built by Carthage to create your own temporary pod, I think.

```
$git clone https://github.com/couchbase/couchbase-lite-ios.git
$cd couchbase-lite-ios
$git submodule update --init --recursive
$carthage update --platform iOS
```

To install and use carthage, https://github.com/Carthage/Carthage.

**Comment by CharlesGrimont** [comment]
The following command fails since there is not any Cartfile included in this project.

    $carthage update --platform iOS

Have I to create a Cartfile myself ? 
And so refactor myself the all project ?

**Comment by pasin** [comment]
Sorry, I gave you the wrong instruction. You don't need to clone couchbase-lite-ios. Do the following instead.

1. First create a Cartfile pointing to couchbase-lite-ios repo. See some example [here](https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#cartfile).
2. Run `$carthage update --platform iOS`

**Comment by djpongh** [comment]
This will be fixed in our 2.0 release.
